desk/lib/bolt.hoon

```diff
@@ -162,72 +160,56 @@
     =^  cards  yosh  (on-fail:ag term tang)
     [cards this]
   --
+  ::
   |%
-  ++  permissions
-    |_  [=bowl:gall state=state-0]
-    +*  parent  (sein:title our.bowl now.bowl src.bowl)
+  ++  is-allowed
+    |=  =bowl:gall
+    ^-  ?
+    ::
+    =*  parent  (sein:title our.bowl now.bowl src.bowl)
+    =*  user    src.bowl
+    =*  moon-gud  |(=(our.bowl parent) (~(has in whitelist) parent))
+    =*  moon-bad  |(=(our.bowl parent) (~(has in blacklist) parent))
+    ::
+    ::  Do not allow if in blacklist or kids and kid of blacklister
+    ?&  |(!(~(has in blacklist) user) &(kids !moon-bad))
+        ::  If whitelist is off -- allow user,
+        ?|  !white
+        ::  If whitelist is on, check if user is in whitelist or kid of whitelister.
+            =(our.bowl user)
+            (~(has in whitelist) user)
+            &(kids moon-gud)
+    ==  ==
   ::
-    ++  is-allowed
-      |=  user=@p
-      =+  state
-      =*  moon-gud  |(=(our.bowl parent) (~(has in users.whitelist) parent))
-      =*  moon-bad  |(=(our.bowl parent) (~(has in users.blacklist) parent))
-      ^-  ?
-      ::
-      ::  Do not allow if in blacklist or kids and kid of blacklister
-      ?&  |(!(~(has in users.blacklist) user) &(kids !moon-bad))
-      ::
-      :: If whitelist is off -- allow user,
-          ?:  =(which %|)
-            %&
```

Comment:
Your boolean expression here is `?:  cond  &  other`. This always simplifies to `?|  cond  other  ==`. Try drawing a truth table on paper to see why.